Dynatrace/dynatrace-operator
# Adds logic for configurable unmounted volume gc

# Description

The log gc in the csi driver is too complicated and not very effective.
As it only deletes folders IF there are (too many log files OR too big is the sum of the log files) AND more 2 weeks old.
This logic is not very helpful when it comes to short living pods, as the folders created by the CSI driver is counted towards the memory usage of the CSI driver.

So now we only care about the age of the folders.
Furthermore now the max age can be configured via a envvar on the provisioner container. (name of envvar: `MAX_UNMOUNTED_VOLUME_AGE`)
Or in the values.yaml incase of Helm.

Some refactor:
- Log garbage collection was misleading, as it did not gc logs per-say, it gc-ed unmounted volumes in case there were too many logs.
  -  Renamed to "unmounted volumes garbage collection", so we say what it actually does.

Why not a feature-flag?
There could be multiple dynakubes with the same tenantUUID affecting the same node. 
- To be able to handle the `MAX_UNMOUNTED_VOLUME_AGE` per dynakube would result in major code modifications. (+ extra confusion from the user side)

## How can this be tested?
Set the `MAX_UNMOUNTED_VOLUME_AGE` on the provisioner container to some value that suits you, check if unmounted volumes are removed on the node after the given time.

The value of `MAX_UNMOUNTED_VOLUME_AGE` can be whatever you `time.ParseDuration` can handle.

## Checklist
